test/e2e/tests/test_version.py

```diff
@@ -231,4 +231,84 @@ def test_smoke_ref(self, lambda_client, lambda_function):
         time.sleep(DELETE_WAIT_AFTER_SECONDS)
 
         # Check alias doesn't exist
-        assert not lambda_validator.version_exists(function_resource_name, version_number)
\ No newline at end of file
+        assert not lambda_validator.version_exists(function_resource_name, version_number)
+    
+    def test_function_event_invoke_config(self, lambda_client, lambda_function):
+        (_, function_resource) = lambda_function
+        lambda_function_name = function_resource["spec"]["name"]
+
+        resource_name = random_suffix_name("lambda-version", 24)
+
+        resources = get_bootstrap_resources()
+        logging.debug(resources)
+
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["AWS_REGION"] = get_region()
+        replacements["FUNCTION_NAME"] = lambda_function_name
+        replacements["VERSION_NAME"] = resource_name
+        replacements["MAXIMUM_EVENT_AGE_IN_SECONDS"] = "100"
+        replacements["MAXIMUM_RETRY_ATTEMPTS"] = "1"
+        replacements["ON_SUCCESS_DESTINATION"] = resources.EICQueueOnSuccess.arn
+        replacements["ON_FAILURE_DESTINATION"] = resources.EICQueueOnFailure.arn
+
+        # Load version CR
+        resource_data = load_lambda_resource(
+            "version_event_invoke_config",
+            additional_replacements=replacements,
+        )
+        logging.debug(resource_data)
+
+        # Create k8s resource
+        ref = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+            resource_name, namespace="default",
+        )
+        k8s.create_custom_resource(ref, resource_data)
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        assert cr is not None
+        assert k8s.get_resource_exists(ref)
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        lambda_validator = LambdaValidator(lambda_client)
+
+        version_number = cr['status']['version']
```

Comment:
Can we assert that the version is 1 ? or 2?

Reply:
Maybe more in the `test_smoke` function

Reply:
I am not sure if that would work or fail, giving it a try by hard coding version to be `1` for `test_smoke` function. My guess is while running tests if the controller calls the same API twice for publishing version, the second time the Version won't be published as the same version_number (which is 1) will be passed.

Reply:
I tried keeping the version to be 1, but the tests failed. My guess is this is because controller might be using same function and publishing Version for it and then deleting it.  The concept is even after deleting the function and its associated Version, when it creates the Function with same name and tries to publish Version for it, the Version number gets incremented from the last published version. Thus we cannot hard bound the Version number, instead we're getting the Version number after the Version is published and then using it to check if the Version exists.

Reply:
Added a new test function named `test_smoke_version_specified` keeping Version to be `1` and testing.